File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    let firstAppear = new Map();
    for(const mapping of keymap){
        [...mapping].forEach((v, idx) => {
            firstAppear.set(v, Math.min((firstAppear.get(v) || 100), idx+1));
        })
    }
    
    let result = [];
    for(const target of targets){
        cnt = 0;
        for(const t of target){
            if(!firstAppear.has(t)){
                cnt = -1;
                break;
            }
            cnt += firstAppear.get(t);
        }
        result.push(cnt);
    }
    return result;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    first_appear = {}
    for mapping in keymap:
        for idx, k in enumerate(mapping):
            if k not in first_appear:
                first_appear[k] = idx+1
            else:
                first_appear[k] = min(first_appear[k], idx+1)

    result = []
    for target in targets:
        cnt = 0
        for t in target:
            if t not in first_appear:
                cnt = -1
                break
            cnt += first_appear[t]
        result.append(cnt)
    
    return result